tslearn/matrix_profile.py

```diff
@@ -130,21 +146,56 @@ def _transform(self, X, y=None):
         n_ts, sz, d = X.shape
         output_size = sz - self.subsequence_length + 1
         X_transformed = np.empty((n_ts, output_size, 1))
-        scaler = TimeSeriesScalerMeanVariance()
-        band_width = int(np.ceil(self.subsequence_length / 4))
-        for i_ts in range(n_ts):
-            segments = _series_to_segments(X[i_ts], self.subsequence_length)
-            if self.scale:
-                segments = scaler.fit_transform(segments)
-            n_segments = segments.shape[0]
-            segments_2d = segments.reshape((-1, self.subsequence_length * d))
-            dists = squareform(pdist(segments_2d, "euclidean"))
-            band = (np.tri(n_segments, n_segments,
-                           band_width, dtype=np.bool) &
-                    ~np.tri(n_segments, n_segments,
-                            -(band_width + 1), dtype=np.bool))
-            dists[band] = np.inf
-            X_transformed[i_ts] = dists.min(axis=1, keepdims=True)
+
+        if self.implementation == "stump":
+            import stumpy
```

Comment:
Instead of importing in the function, I'd rather have a mechanism like the following at the top of the module:

https://github.com/tslearn-team/tslearn/blob/0d4c4ce5391f00aa68769e0ad81bccd0b1c32354/tslearn/bases.py#L9-L23